### ISSUE TITLE
[Bug]: ImportError: cannot import name 'LinearLayer' from 'llama_index.finetuning.embeddings.adapter_utils'

### ISSUE THREAD

**Issue by SAAD-3XK** [run-llama/llama_index#10743]
### Bug Description

For some reason I am getting this Import error when I attempt to initialize the `EmbeddingAdapterFinetuneEngine` on top of a HuggingFace embedding model and train_dataset. The code is taken directly from the documentation and I am running the latest version.

The only thing I've changed is how I set the base_embed_model. Instead of using `resolve_embed_model` (which also leads to an import error):

```
from llama_index.finetuning import EmbeddingAdapterFinetuneEngine
from llama_index.core.embeddings import resolve_embed_model
import torch

base_embed_model = resolve_embed_model("local:BAAI/bge-small-en")
```

I set it like this:

`base_embed_model = "BAAI/bge-base-en-v1.5"`

And run into an import error when I run the following:

```
finetune_engine = EmbeddingAdapterFinetuneEngine(
    train_dataset,
    base_embed_model,
    model_output_path="model_v3_finetune",
    epochs=5,
    verbose=True,
)
```

### Version

0.10.4

### Steps to Reproduce

Create train and val nodes using SentenceSplitter
```
from llama_index.core.node_parser import SentenceSplitter
from llama_index.readers.file import PyMuPDFReader

file_path = 'path/to/doc.pdf'
loader = PyMuPDFReader()
docs=loader.load(file_path)
parser = SentenceSplitter()
nodes = parser.get_nodes_from_documents(docs)
```

Generate train and val datasets:
```
from llama_index.finetuning import (
    generate_qa_embedding_pairs,
    EmbeddingQAFinetuneDataset,
)

train_dataset = generate_qa_embedding_pairs(llm=OpenAI(model="gpt-3.5-turbo-0125"),nodes=train_nodes, num_questions_per_chunk=2)
train_dataset.save_json("train_dataset.json")

val_dataset = generate_qa_embedding_pairs(llm=OpenAI(model="gpt-3.5-turbo-0125"), nodes=val_nodes, num_questions_per_chunk=2)
val_dataset.save_json("val_dataset.json")
```

Initialize the finetuning engine:
```
from llama_index.finetuning import EmbeddingAdapterFinetuneEngine

base_embed_model = "BAAI/bge-base-en-v1.5"

finetune_engine = EmbeddingAdapterFinetuneEngine(
    train_dataset,
    base_embed_model,
    model_output_path="model_v3_finetune",
    epochs=5,
    verbose=True,
)
```

### Relevant Logs/Tracbacks

```shell
---------------------------------------------------------------------------
ImportError                               Traceback (most recent call last)
Cell In[6], line 7
      2 # from llama_index.core.embeddings import resolve_embed_model
      3 # import torch
      5 base_embed_model = "BAAI/bge-base-en-v1.5"
----> 7 finetune_engine = EmbeddingAdapterFinetuneEngine(
      8     train_dataset,
      9     base_embed_model,
     10     model_output_path="model_v3_finetune",
     11     # bias=True,
     12     epochs=5,
     13     verbose=True,
     14     # optimizer_class=torch.optim.SGD,
     15     # optimizer_params={"lr": 0.01}
     16 )

File ~/Personal/LlamIndex/llama-env/lib/python3.10/site-packages/llama_index/finetuning/embeddings/adapter.py:53, in EmbeddingAdapterFinetuneEngine.__init__(self, dataset, embed_model, batch_size, epochs, adapter_model, dim, device, model_output_path, model_checkpoint_path, checkpoint_save_steps, verbose, bias, **train_kwargs)
     51 """Init params."""
     52 import torch
---> 53 from llama_index.finetuning.embeddings.adapter_utils import (
     54     BaseAdapter,
     55     LinearLayer,
     56 )
     58 self.dataset = dataset
     59 self.embed_model = embed_model

ImportError: cannot import name 'LinearLayer' from 'llama_index.finetuning.embeddings.adapter_utils' (/home/saad/Personal/LlamIndex/llama-env/lib/python3.10/site-packages/llama_index/finetuning/embeddings/adapter_utils.py)
```

**Comment by nerdai** [comment]
Hi @SAAD-3XK, sorry you're running into some trouble here with v0.10. I'm here to help, and a longer message will follow shortly. I can confirm tho that the `resolve_embed_model` import is not working, which perplexes me. We can directly import it from `llama_index.core.embeddings.utils` but the shortcut import should work too.

**Comment by nerdai** [comment]
Okay @SAAD-3XK, part 2:

As you know the fine-tuning module has been ripped out and now is its own Python package. Additionally, the `embeddings.adapter` module has also been separated from core and it too is its own package. And, from the looks of it, it doesn't look like the `embeddings.adapter` module is referencing the correct import path when trying to import `LinearLayer`:

Currently from `llama_index.finetuning.embeddings.adapter`:
```python
...
        from llama_index.finetuning.embeddings.adapter_utils import (
            BaseAdapter,
            LinearLayer,
        )
...
```

Which should be `from llama_index.embeddings.adapter.utils`. Will submit a PR that fixes this shortly.

**Comment by nerdai** [comment]
This should be fixed now and was included in this [PR](https://github.com/run-llama/llama_index/pull/10770/files). And, the fix should be available in the latest version (v0.1.2) of `llama-index-finetuning`. Closing this as a result, but please re-open if this did not resolve everything. 🙏
